### ISSUE TITLE
AppCenter unable to process payments

### ISSUE THREAD

**Issue by thaddeusc1** [elementary/appcenter#1827]
### What Happened?

*tl;dr*: ~~AppCenter's~~ QuickWord's API key for Stripe has **expired**.

## What is happening
Attempting to make a payment through AppCenter returns the error, "There Was a Problem Processing Your Payment."
![Screenshot from 2022-03-29 10 38 52@2x](https://user-images.githubusercontent.com/19389910/160638556-f6f84fb8-e8fc-4f19-bfd5-890fed5ca57d.jpeg)

### Steps to Reproduce

1. Run AppCenter with debug output—i.e.,
    - `G_MESSAGES_DEBUG=all io.elementary.appcenter`.
2. Attempt to perform *any* action that would process a payment from the user—e.g.,
    1.  Search AppCenter for the app named "[QuickWord](https://appcenter.elementary.io/com.github.hezral.quickword)".

    2. From the search results, select the QuickWord app.

    4. Navigate to the *bottom* of QuickWord's product page in AppCenter.

    5. Click on "Fund".
    6. Enter payment information.
    7. Press the *Pay* button.
 

### Expected Behavior

When a user attempts to authorize a payment via AppCenter, the payment should be successfully processed.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.942: CLIENT[0x55ece32a9210]: Starting synchronous TLS handshake
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.942: CLIENT[0x55ece32a9210]: TLS handshake thread starts
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.942: CLIENT[0x55ece32a9210]: claiming operation OP_HANDSHAKE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.942: CLIENT[0x55ece32a9210]: claiming operation OP_HANDSHAKE succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.950: CLIENT[0x55ece32a9210]: verifying peer certificate
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: TLS handshake thread succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: synchronous TLS handshake thread completed
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: finishing TLS handshake
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: TLS handshake has finished successfully
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: yielding operation OP_HANDSHAKE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: starting to write 307 bytes to TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_WRITE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_WRITE succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: yielding operation OP_WRITE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: successfully write 307 bytes to TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: starting to write 303 bytes to TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_WRITE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_WRITE succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: yielding operation OP_WRITE
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: successfully write 303 bytes to TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: starting to read data from TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:23.960: CLIENT[0x55ece32a9210]: claiming operation OP_READ succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: yielding operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: successfully read 1005 bytes from TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: starting to read data from TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_READ succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: yielding operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: reading data from TLS connection has failed: WOULD_BLOCK
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: starting to read data from TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_READ succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: yielding operation OP_READ
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: reading data from TLS connection has failed: WOULD_BLOCK
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: starting to close the TLS connection
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_CLOSE_BOTH
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: claiming operation OP_CLOSE_BOTH succeeded
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: yielding operation OP_CLOSE_BOTH
(io.elementary.appcenter:5230): GLib-Net-DEBUG: 10:38:24.058: CLIENT[0x55ece32a9210]: the TLS connection has been closed successfully
** (io.elementary.appcenter:5230): DEBUG: 10:38:24.058: StripeDialog.vala:449: Stripe data:{
  "error": {
    "code": "platform_api_key_expired",
    "doc_url": "https://stripe.com/docs/error-codes/platform-api-key-expired",
    "message": "Expired API Key provided: pk_live_*********************************************************************************************l3tXzb. Platform access may have been revoked.",
    "type": "invalid_request_error"
  }
}

** (io.elementary.appcenter:5230): DEBUG: 10:38:24.058: StripeDialog.vala:544: Stripe error type: invalid_request_error
```

### Intel Core i9-11980HK @ 4.90GHz
```
    Core Count:           8                                                 
    Thread Count:         16                                                
    Extensions:           SSE 4.2                                           
                          + AVX512_VNNI                                     
                          + AVX512CD                                        
                          + AVX2                                            
                          + AVX                                             
                          + RDRAND                                          
                          + FSGSBASE                                        
    Cache Size:           24 MB                                             
    Microcode:            0x3c                                              
    Core Family:          Tiger Lake                                        
    Scaling Driver:       intel_pstate powersave (EPP: balance_performance) 
```
### NVIDIA GeForce RTX 3080 Laptop GPU
```
    Frequency:            1545/7000MHz             
    BAR1 / Visible vRAM:  16384 MiB                
    OpenGL:               4.6.0                    
    Vulkan:               1.3.194                  
    OpenCL:               OpenCL 3.0 CUDA 11.6.110 
    Display Driver:       NVIDIA 510.54            
    Screen:               3840x2160                
```
### Alienware x17 R1
```
    BIOS Version:         1.8.0                                   
    Chipset:              Intel HM570                        
    Audio:                Realtek ALC3281                       
    Network:              Killer E3100G + Killer 1675w 
```
## Memory
64GB DDR4-3200AA

### Western Digital WD_BLACK SN770 1TB
```
    File-System:          ext4                         
    Mount Options:        errors=remount-ro noatime rw 
    Disk Scheduler:       NONE                         
    Disk Details:         Block Size: 4096             
```
### Micron 3400 NVMe 2048GB
```
    File-System:         NTFS
    Mount Options:       Not mounted
```
### Elementary 6.1
```
    Kernel:               5.13.0-37-generic (x86_64)                                                           
    Display Server:       X Server 1.20.13                                                                     
    Compiler:             GCC 10.3.0 + Clang 10.0.0-4ubuntu1                                                   
    Security:             itlb_multihit: Not affected                                                          
                          + l1tf: Not affected                                                                 
                          + mds: Not affected                                                                  
                          + meltdown: Not affected                                                             
                          + spec_store_bypass: Mitigation of SSB disabled via prctl and seccomp                
                          + spectre_v1: Mitigation of usercopy/swapgs barriers and __user pointer sanitization 
                          + spectre_v2: Mitigation of Enhanced IBRS IBPB: conditional RSB filling              
                          + srbds: Not affected                                                                
                          + tsx_async_abort: Not affected
```

**Comment by thaddeusc1** [comment]
Based on the [latest version of AppCenter](https://github.com/elementary/appcenter/tree/3.9.1)'s workflow for app payments, this issue needs to be fixed downstream in QuickWord.
- Closing this issue, as no change to AppCenter is *required* to resolve this issue.

**Comment by janxkoci** [comment]
Today I tried to pay for **three different apps** from different devs - and all failed. I thought it's more likely the problem is in the AppCenter itself rather than all three apps having issue with their payment setups.

@thaddeusc1 How did you get the log from Appcenter, please? I'd like to check that the problem is really at the app dev end(s) and not somewhere else...

**Comment by janxkoci** [comment]
So I've finally found another app I wanted to pay for and tried the above trick with debug info. Didn't help. The payment failed and the output gives no clue why. Here is the entire (very short) output:

```bash
$ G_MESSAGES_DEBUG=all io.elementary.appcenter
(process:7549): GLib-GIO-DEBUG: 14:51:57.652: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ?gsettings-backend?
(process:7549): dconf-DEBUG: 14:51:57.653: watch_fast: "/io/elementary/appcenter/settings/" (establishing: 0, active: 0)
(process:7549): GLib-DEBUG: 14:51:57.653: unsetenv() is not thread-safe and should not be used after threads are created
** (process:7549): DEBUG: 14:51:57.654: ScreenshotCache.vala:34: screenshot path is at /home/jena/.cache/io.elementary.appcenter/screenshots
(process:7549): dconf-DEBUG: 14:51:57.657: watch_established: "/io/elementary/appcenter/settings/" (establishing: 1)
(process:7549): GLib-GIO-DEBUG: 14:51:57.660: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
(io.elementary.appcenter:7549): dconf-DEBUG: 14:51:57.671: sync
** (io.elementary.appcenter:7549): DEBUG: 14:51:57.671: ScreenshotCache.vala:71: Screenshot folder size is 77,4 MB
```

This appeared even before the AppCenter window opened, no new output was produced during the payment (or any other action I performed).

Now what? :shrug:
